### PR TITLE
possibility to specify custom mock names

### PR DESCRIPTION
**Pull request by mariusstaicu**

### CODE REVIEW

**Comment by mariusstaicu** [comment]
I signed it!

**Comment by mariusstaicu** [comment]
I signed it!

**Comment by vborcea** [comment]
nice work

File: mockgen/mockgen.go [hunk 1]
```diff
@@ -98,13 +99,28 @@ func main() {
...
 	if _, err := dst.Write(g.Output()); err != nil {
 		log.Fatalf("Failed writing to destination: %v", err)
 	}
 }
+func parseMockNames(names string) map[string]string {
+	mocksMap := make(map[string]string)
+	for _, kv := range strings.Split(names, ",") {
+		parts := strings.SplitN(kv, "=", 2)
+		if len(parts) != 2 || parts[1] == "" {
+			log.Fatalf("bad mock names spec: %v", kv)
```
Comment (vborcea) [inline mockgen/mockgen.go]: If the parts are not OK you should continue with the next part and drop the current "bad" ones

Comment (mariusstaicu) [reply mockgen/mockgen.go]: I would rather exit with error if that happens.

Comment (vborcea) [reply mockgen/mockgen.go]: Ok

**Comment by mariusstaicu** [comment]
I signed it!

**Pull request merged**